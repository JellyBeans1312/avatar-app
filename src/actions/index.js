export const showCharacters = characters => ({
  type: 'SHOW_CHARACTERS',
  characters
})