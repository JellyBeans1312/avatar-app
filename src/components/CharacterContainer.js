import React from 'react';

const CharacterContainer = () => {
  return (
    'wassup'
  )
}

export default CharacterContainer;