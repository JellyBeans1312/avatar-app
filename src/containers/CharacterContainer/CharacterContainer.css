.allCharacters {
  display: flex;
  flex-wrap: wrap;
}

.symbols {
  margin: 0 50px;
  height: 150px;
}

.title {
  margin: auto;
}

.categories {
  height: 30vh;
  margin: 15px
}

.all {
  display: inline-block;
}