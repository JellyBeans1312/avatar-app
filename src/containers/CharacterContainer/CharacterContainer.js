import React from 'react';
import { connect } from 'react-redux';
import { filterCharacters } from '../../actions';
import { Card } from '../Card/Card'
import water from '../../images/water-symbol.png'
import fire from '../../images/fire-symbol.png'
import earth from '../../images/earth-symbol.png'
import air from '../../images/air-symbol.png';
import { Redirect } from 'react-router-dom';



export const CharacterContainer = ({characters, filterStatus, filterClick}) => {

  const allCharacters = () => {
    return characters.map(character => {
      return <Card character={character} key={character.id} />;
    });
  };

  const waterNation = () => {
    let water = characters.filter(character => {
      const aff = character.affiliation
      if (aff === undefined) {
        return null
      } else {
        return aff.includes('Water')
      }
    });
    return water.map(character => {
      return <Card character={character} key={character.id} affiliation='water'/>;
    });
  };

  const fireNation = () => {
    let fire = characters.filter(character => {
      const aff = character.affiliation
      if (aff === undefined) {
        return null
      } else {
        return aff.includes('Fire')
      }
    });
    return fire.map(character => {
      return <Card character={character} key={character.id} />;
    });
  };

  const earthNation = () => {
    let earth = characters.filter(character => {
      const aff = character.affiliation
      if (aff === undefined) {
        return null
      } else {
        return aff.includes('Earth')
      }
  });
    return earth.map(character => {
      return <Card character={character} key={character.id} />;
    });
  };

  const airNation = () => {
    let air = characters.filter(character => {
      const aff = character.affiliation
      if (aff === undefined) {
        return null
      } else {
        return aff.includes('Air')
      }
    });
    return air.map(character => {
      return <Card character={character} key={character.id} />;
    });
  };

  const handleClick = e => {
    e.preventDefault()
    const filter = e.target.id;
    filterClick(filter);
  };

  return (
    <>
      <header className='header'>
        <h2 className='title'>
          Pick a Nation!!
        </h2>
        <div className='categories'>
          <button className=''id='all' onClick={e => handleClick(e)}>
            Show all
          </button>
          <img src={water} alt='water symbol'className='' id='water' onClick={e => handleClick(e)}/>

          <img src={fire} alt='fire symbol'className='' id='fire' onClick={e => handleClick(e)}>
            {/* fire */}
          </img >
          <img src={earth} alt='earth symbol'className='' id='earth' onClick={e => handleClick(e)}>
            {/* earth */}
          </img >
          <img src={air} alt='air symbol'className='' id='air' onClick={e => handleClick(e)}>
            {/* air */}
          </img >
        </div>
      </header>

      {filterStatus === 'all' && (
        <section className='all-characters'>{allCharacters()}</section>
      )}
      {filterStatus === 'water' && (
        <section className='water-characters'>{waterNation()}</section>
      )}
      {filterStatus === 'fire' && (
        <section className='fire-characters'>{fireNation()}</section>
      )}
      {filterStatus === 'earth' && (
        <section className='earth-characters'>{earthNation()}</section>
      )}
      {filterStatus === 'air' && (
        <section className='air-characters'>{airNation()}</section>
      )}
    </>
  );
};

export const mapStateToProps = (store) => ({
  characters: store.characters,
  filterStatus: store.filter
});

export const mapDispatchToProps = dispatch => ({
  filterClick: filter => dispatch(filterCharacters(filter))
});



export default connect(mapStateToProps, mapDispatchToProps)(CharacterContainer);