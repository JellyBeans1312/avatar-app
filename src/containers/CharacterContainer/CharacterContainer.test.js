import React from 'react';
import { shallow } from 'enzyme';
import { CharacterContainer, mapStateToProps, mapDispatchToProps } from './CharacterContainer';
import { filterCharacters } from '../../actions'

describe('filters and methods', () => {
  let wrapper;
  let mockEvent;
  let allChars
  let props
  beforeEach(() => {
    mockEvent = {
      target: {
        id: 1
      }
    };
    allChars = [{ id: 1, name: 'Aang', affilitaion: 'air' }, 
    { id: 2, name: 'Katara', affilitaion: 'water'},
    { id: 3, name: 'Toph', affilitaion: 'earth'},
    { id: 4, name: 'Zuko', affilitaion: 'fire'},

  ];
    props = {
      characters: allChars,
      filterStatus: 'all',
      filterClick: jest.fn()
    };
    wrapper = shallow(<CharacterContainer {...props} />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it.skip('should iterate through all characters', () => {
    
    expect(wrapper.instance().allCharacters().length).toEqual(4);
  });

  it.skip('should filter through all characters and only return the people from water nation', () => {
    expect(wrapper.instance().waterNation().length).toEqual(1);
  });

  it.skip('should filter through all characters and only return the people from fire nation', () => {
    expect(wrapper.instance().fireNation().length).toEqual(1);
  });
  it.skip('should filter through all characters and only return the people from earth nation', () => {
    expect(wrapper.instance().earthNation().length).toEqual(1);
  });

  it.skip('should filter through all characters and only return the people from air nation', () => {
    expect(wrapper.instance().airNation().length).toEqual(1);
  });
});

describe('mapStateToProps', () => {
  it('should contain the characters and filterStatus', () => {
    const mockState = {
      characters: 'hello',
      filter: 'all'
    };
    const expected = {
      characters: 'hello',
      filterStatus: 'all'
    };
    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps).toEqual(expected);
  });
});

describe('mapDispatchToProps', () => {
  it('should map filterCharacters to props', () => {
    const mockDispatch = jest.fn();
    const actionToDispatch = filterCharacters(undefined);
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.filterClick();
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });
});
