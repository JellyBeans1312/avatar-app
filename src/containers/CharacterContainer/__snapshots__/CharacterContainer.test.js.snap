// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filters and methods should match the snapshot 1`] = `ShallowWrapper {}`;
